server/src/main/java/org/apache/druid/server/coordinator/duty/BalanceSegments.java

```diff
@@ -145,20 +146,41 @@ private void balanceTier(
         params.getCoordinatorDynamicConfig().getDecommissioningMaxPercentOfMaxSegmentsToMove();
     int maxSegmentsToMoveFromDecommissioningNodes =
         (int) Math.ceil(maxSegmentsToMove * (decommissioningMaxPercentOfMaxSegmentsToMove / 100.0));
+
+    // An aggregate result that accumulates during decommissioning moves and guild replication moves
+    Pair<Integer, Integer> specialtyMoveResult;
+
     log.info(
         "Processing %d segments for moving from decommissioning servers",
         maxSegmentsToMoveFromDecommissioningNodes
     );
     Pair<Integer, Integer> decommissioningResult =
-        balanceServers(params, decommissioningServers, activeServers, maxSegmentsToMoveFromDecommissioningNodes);
+        balanceServers(params, decommissioningServers, activeServers, maxSegmentsToMoveFromDecommissioningNodes, false);
+
+    // If guildReplication is enabled, run balanceSegments focused solely on moving segments who live on <= 1 guild.
```

Comment:
this comment is not fully accurate. just cuz guildReplication is enabled doesn't cause this to run. the coordinator dynamic config needs to be > 0 as well.